src/devices/neo.ts

```diff
@@ -154,96 +154,118 @@ const definitions: Definition[] = [
+        toZigbee: [tuya.tz.datapoints],
         onEvent: tuya.onEventSetTime,
         configure: tuya.configureMagicPacket,
         exposes: [
             e.switch(),
-            e.enum('status', ea.STATE, ['off', 'auto', 'disabled']).withDescription('Status'),
-            e.numeric('countdown', ea.STATE_SET).withUnit('min').withValueMin(1).withValueMax(240).withDescription('Countdown'),
-            e.numeric('countdown_left', ea.STATE).withUnit('min').withValueMin(1).withValueMax(240).withDescription('Countdown left'),
-            e
-                .numeric('water_current', ea.STATE)
-                .withUnit('L/min')
-                .withValueMin(0)
-                .withValueMax(3785.41)
-                .withValueStep(0.001)
-                .withDescription('Current water flow (L/min)'),
-            e.numeric('battery_percentage', ea.STATE).withUnit('%').withValueMin(0).withValueMax(100).withDescription('Battery percentage'),
-            e
-                .numeric('water_total', ea.STATE)
-                .withUnit('L')
-                .withValueMin(0)
-                .withValueMax(378541.0)
-                .withValueStep(0.001)
-                .withDescription('Total water flow (L)'),
-            e.binary('fault', ea.STATE, 'DETECTED', 'NOT_DETECTED').withDescription('Fault status'),
-            e.enum('weather_delay', ea.STATE_SET, ['24h', '48h', '72h', 'cancel']).withDescription('Weather delay'),
-            e.text('normal_timer', ea.STATE_SET).withDescription('Normal timer'),
-            e.binary('switch_enabled', ea.STATE_SET, 'ON', 'OFF').withDescription('Switch enabled'),
-            e.numeric('smart_irrigation', ea.STATE).withDescription('Smart irrigation'),
-            e.binary('total_flow_reset_switch', ea.STATE_SET, 'ON', 'OFF').withDescription('Total flow reset switch'),
-            e
-                .numeric('quantitative_watering', ea.STATE_SET)
-                .withUnit('L')
-                .withValueMin(0)
-                .withValueMax(10000)
-                .withDescription('Quantitative watering'),
-            e.binary('flow_switch', ea.STATE_SET, 'ON', 'OFF').withDescription('Flow switch'),
+            e.enum('status', ea.STATE, ['off', 'auto', 'disabled', 'app_manual', 'key_control']).withDescription('Status'),
+            e.numeric('countdown', ea.STATE_SET).withUnit('min').withValueMin(1).withValueMax(60).withDescription('Count down'),
+            e.numeric('countdown_left', ea.STATE).withUnit('min').withValueMin(1).withValueMax(60).withDescription('Countdown left time'),
+            e.numeric('water_total', ea.STATE).withUnit('L').withValueMin(0).withValueStep(0.001).withDescription('Water total (L)'),
+            e.numeric('water_current', ea.STATE).withUnit('L/min').withValueMin(0).withValueStep(0.001).withDescription('Current water flow (L/min)'),
+            e.binary('current_switch', ea.STATE_SET, 'ON', 'OFF').withDescription('Flow switch'),
```

Comment:
ahhhh maybe it was the reason that flow switch wasn't work for me, the current implementation it has `flow_switch` instead of `current_switch`